### DISCUSSION TITLE
How to configure Custom Embedding Provider

### DISCUSSION THREAD

**Discussion by NirnayK** [NVIDIA-NeMo/Guardrails#557] [Q&A] [answered]
In Custom Embedding Provider For the function: async def search(self, text: str, max_results: int = 20) -> List[IndexItem]: 

Can anyone explain what needs to be returned in an IndexItem
what should be returned in the text attribute and what should be returned in meta attribute.

**Comment by NirnayK** [comment]
Is there any code with a fully implemented Custom Embedding Provider ?

**Reply by drazvan** [reply, marked as answer]
@NirnayK : the default implementation is a good reference point: https://github.com/NVIDIA/NeMo-Guardrails/blob/main/nemoguardrails/embeddings/basic.py. 

And to respond to your questions:
1. The `text` is the text of the chunk. 
2. In the meta it's a dict with different types of info depending on the purpose of the index (e.g., `body` and `title` for the kb; `intent` for the user utterances).
